core/p2p/p2pv2/tls.go

```diff
@@ -0,0 +1,153 @@
+package p2pv2
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/sec"
+	"io/ioutil"
+	"net"
+)
+
+// ID is the protocol ID (used when negotiating with multistream)
+const ID = "/tls/1.0.0"
+
+// Transport constructs secure communication sessions for a peer.
+type Transport struct {
+	config *tls.Config
+
+	privKey   crypto.PrivKey
+	localPeer peer.ID
+}
+
+var _ sec.SecureTransport = &Transport{}
+
+func New(path, serviceName string) func(key crypto.PrivKey) (*Transport, error) {
+	return func(key crypto.PrivKey) (*Transport, error) {
+		bs, err := ioutil.ReadFile(path + "/cacert.pem")
```

Comment:
It is recommended to use `filepath.Join`, which can handle some corner case and better cross-platform. Same in other places.